ride/gwx_reward.ride

```diff
@@ -500,21 +501,30 @@ func calcD(
     ([], zeroBigInt.toString())
   } else {
     let ann = amp * nCoins
-    let arr = [1, 2, 3, 4, 5, 6, 7]
-    func calc(acc: (BigInt, BigInt|Unit), cur: Int) = {
-      let (d, dPrev) = acc
-      # dp0 = d
-      # dp1 = dp0 * d / (x1 * nCoins)
-      # dp2 = dp1 * d / (x2 * nCoins) = (dp0 * d / (x1 * nCoins)) * d / (x2 * nCoins) = d^3 / (x1 * x2 * nCoins^2)
-      let dp = d * d * d / (x1 * x2 * nCoins * nCoins)
-      let dNext = (ann * s / aPrecision + dp * nCoins) * d / ((ann - aPrecision) * d / aPrecision + (nCoins + 1.toBigInt()) * dp)
-      (dNext, d)
+    let arr = [0, 1, 2, 3, 4, 5, 6]
+    func calc(acc: (BigInt, BigInt|Unit, Int|Unit), cur: Int) = {
+      let (d, dPrev, found) = acc
+      if (found != unit) then acc else {
+        # dp0 = d
+        # dp1 = dp0 * d / (x1 * nCoins)
+        # dp2 = dp1 * d / (x2 * nCoins) = (dp0 * d / (x1 * nCoins)) * d / (x2 * nCoins) = d^3 / (x1 * x2 * nCoins^2)
+        let dp = d * d * d / (x1 * x2 * nCoins * nCoins)
+        let dNext = (ann * s / aPrecision + dp * nCoins) * d / ((ann - aPrecision) * d / aPrecision + (nCoins + 1.toBigInt()) * dp)
+        let dDiff = absBigInt(dNext - d.value())
+        if (dDiff <= targetPrecision) then {
+          (dNext, d, cur)
+        } else {
+          (dNext, d, unit)
+        }
+      }
     }
-    let (dNext, dPrev) = FOLD<7>(arr, (s, unit), calc)
-    let dDiff = abs(dNext - dPrev.value())
-    if (dDiff <= targetPrecision) then {
+    let (dNext, dPrev, found) = FOLD<7>(arr, (s, unit, unit), calc)
```

Comment:
Why only 7 iterations?

Now we have fast return in fold function which does not use much complexity.